semgrep/sol-rules.t.sol

```diff
@@ -0,0 +1,403 @@
+// Semgrep tests for Solidity rules are defined in this file.
+// Semgrep tests do not need to be valid Solidity code but should be syntactically correct so that
```

Comment:
Will semgrep fail to parse this and CI fail if there’s a syntax error? I’m wondering if we should run forge fmt on this file as part of CI, since that will enforce syntactically correct code here

Reply:
Semgrep ignores everything inside of the `semgrep` folder unless you explicitly run the tests, so this won't cause any issues in CI.

Reply:
I meant to make sure the solidity code here is actually representative of what we want to test, whereas if there are syntax errors it will not be. So we’d want to fail CI if this file has invalid solidity syntax

Reply:
Ok I added a `forge fmt` check here